unix_integration/src/resolver.rs

```diff
@@ -1054,6 +1057,28 @@ where
                         // AuthCredHandler::MFA is invalid for offline auth
                         return Err(());
                     }
+                    (&mut AuthCredHandler::SetupPin, _) => {
+                        // AuthCredHandler::SetupPin is invalid for offline auth
+                        return Err(());
+                    }
+                    (&mut AuthCredHandler::Pin, PamAuthRequest::Pin { cred }) => {
+                        // The Pin acts as a single device password, and can be cached for offline
```

Comment:
Do we actually want to cache it? This uses our built-in password cache mechanism, rather than calling the backend with the TPM which should still be present and working even offline. Perhaps this is a time where we actually want to uncomment src/idprovider/interface.rs line 165. I think in the case we have an offline auth, we should be using the TPM here for validation of the PIN, rather than our baked in cache mechanism (as convenient as it is). 

Similar, it probably means when you return from unix_user_online_auth_step, if we are using a PIN then we should set AuthCacheAction::None rather than bubbling up the PIN to be cached in the resolver here.

Reply:
Hrm, I didn't consider this.

If we are offline though, providing the PIN would just unlock the TPM key. Ah, I see, you mean uncomment `unix_user_offline_auth_step()`? Then I could just unlock the tpm key and count that as a successful auth?

Reply:
I think my latest changes implement what you've suggested (and I like this better than caching the pin).

Reply:
Yep, that's exactly what I meant! I think I prefer it more than caching the PIN too, because it means we keep everything nice and contained to the TPM and it's controls :)